db/db_properties_test.cc

```diff
@@ -1713,6 +1716,44 @@ TEST_F(DBPropertiesTest, BlockCacheProperties) {
   ASSERT_EQ(0, value);
 }
 
+TEST_F(DBPropertiesTest, TableCacheProperties) {
+  Options options;
+  uint64_t value, new_value;
+
+  options.env = CurrentOptions().env;
+
+  // Block cache properties are not available if block cache is not used.
+  BlockBasedTableOptions table_options;
+  options.table_factory.reset(NewBlockBasedTableFactory(table_options));
+  Reopen(options);
+
+  //
+  // test table_cache access is "live"
+  //  get default max_open_files
+  //
+  ASSERT_TRUE(db_->GetIntProperty(DB::Properties::kTableCacheCapacity, &value));
+  new_value = value / 2;
+
+  std::unordered_map<std::string, std::string> new_options;
+  new_options.insert(std::pair<std::string, std::string>(
+      "max_open_files", std::to_string(new_value)));
+  rocksdb::Status stat = db_->SetDBOptions(new_options);
+  ASSERT_TRUE(stat.ok());
+  ASSERT_TRUE(db_->GetIntProperty(DB::Properties::kTableCacheCapacity, &value));
+  // rocksdb does a -10 to number passed in
+  ASSERT_EQ(new_value - 10, value);
+
+  //
+  // create a new table to see if usage is "live"
```

Comment:
I do not see where this is creating a "new table".  Can you elaborate in the comment?

Reply:
Done.